include/secp256k1_bulletproofs.h

```diff
@@ -0,0 +1,144 @@
+ *  In/Out: data_len: the length of the `data` buffer. Should be initially set to at
+ *                    least 33 times the number of generators; will be set to 33 times
+ *                    the number of generators on successful return (cannot be NULL)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_bulletproofs_generators_serialize(
+    const secp256k1_context* ctx,
+    secp256k1_bulletproofs_generators* gen,
+    unsigned char* data,
+    size_t *data_len
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2);
+
+/** Destroys a list of NUMS generators, freeing allocated memory
+ *  Args:   ctx: pointer to a context object (cannot be NULL)
+ *          gen: pointer to the generator set to be destroyed
+ */
+SECP256K1_API void secp256k1_bulletproofs_generators_destroy(
+    const secp256k1_context* ctx,
+    secp256k1_bulletproofs_generators* gen
+) SECP256K1_ARG_NONNULL(1);
+
+/** Returns the serialized size of an uncompressed proof of a given number of bits
+ *  Args:   ctx: pointer to a context object (cannot be NULL)
+ *  In:  n_bits: number of bits to prove (max 64, should usually be 64)
+ */
+SECP256K1_API size_t secp256k1_bulletproofs_rangeproof_uncompressed_proof_length(
+    const secp256k1_context* ctx,
+    size_t n_bits
+) SECP256K1_ARG_NONNULL(1);
+
+/** Produces an uncompressed rangeproof. Returns 1 on success, 0 on failure.
+ *  Args:   ctx: pointer to a context object (cannot be NULL)
+ *               gens: pointer to the generator set to use, which must have 2*n_bits generators (cannot be NULL)
+ *    asset_gen: pointer to the asset generator for the CT commitment (cannot be NULL)
+ *  Out:      proof: pointer to a byte array to output the proof into
+ * In/OUt:  plen: pointer to the size of the above array; will be set to the actual size of
+ *              the serialized proof. To learn this value in advance, to allocate a sufficient
+ *              buffer, call `secp256k1_bulletproofs_rangeproof_uncompressed_proof_length` or
+ *              use `SECP256K1_BULLETPROOFS_RANGEPROOF_UNCOMPRESSED_MAX_LENGTH`
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_bulletproofs_rangeproof_uncompressed_prove(
+    const secp256k1_context* ctx,
+    const secp256k1_bulletproofs_generators* gens,
+    const secp256k1_generator* asset_gen,
+    unsigned char* proof,
+    size_t* plen,
+    const size_t n_bits,
+    const size_t value,
+    const size_t min_value,
+    const secp256k1_pedersen_commitment* commit,
+    const unsigned char* blind,
+    const unsigned char* nonce,
+    const unsigned char* enc_data,
+    const unsigned char* extra_commit,
+    size_t extra_commit_len
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5) SECP256K1_ARG_NONNULL(10) SECP256K1_ARG_NONNULL(11);
+
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_bulletproofs_rangeproof_uncompressed_verify(
```

Comment:
Comment: Should we add API documentation for this one?

Reply:
Heh, yeah, and we should fix/complete the API docs for the prover.